src/runtime/particle.ts

```diff
@@ -50,18 +52,39 @@ export class Particle {
     if (this.spec.inputs.length === 0) {
       this.extraData = true;
     }
+    this.ready = false;
+    this.created = false;
+  }
+
+  callOnCreate(): void {
+    if (this.created) return;
+    this.created = true;
+    this.onCreate();
   }
 
   /**
-   * Called after handles are synced, override to provide initial processing.
+   * Called after handles are writable, only on first initialization of particle.
    */
-  protected ready(): void {
+  protected onCreate(): void {}
+
+  callOnReady(): void {
```

Comment:
Do we expect `callOnReady()` to be called multiple times?

If yes, why? It's not clear from the calling code.

If not, it could be vastly simplified:
```
callOnReady(): void {
  if (!this.created) this.onCreate();
  this.onReady();
}
```

Reply:
What about this comment?